apps/hrbfc/hrbfc.star

```diff
@@ -0,0 +1,383 @@
+	"description": "Hampton & Richmond Borough Football Club Ground (Beveree Stadium)",
+	"locality": "Hampton",
+	"place_id": "ChIJq4On8EULdkgRYjjDaPfI-SQ",
+	"timezone": "Europe/London"
+}
+"""
+
+HRBFC_TEAM_ID = "4680"
+
+BASE_API_URL = "https://26vyvv1b53.execute-api.us-east-1.amazonaws.com"
+GAME_DATA_PATH = BASE_API_URL + "/getCurrentOrNextGame"
+TEAM_NAMES_PATH = BASE_API_URL + "/getTeamNames"
+
+BACKGROUND_IMAGE_URL = "https://hrbfc-png.s3.amazonaws.com/hrbfc-new-darken.png"
+
+FONT = "tb-8"
+
+GAME_STATUSES = {
+    "TBD": ("TBD", "Time To Be Defined"),
+    "NS": ("Not Started", "Not Started"),
+    "1H": ("First Half", "First Half, Kick Off"),
+    "HT": ("Halftime", "Halftime"),
+    "2H": ("Second Half", "Second Half, 2nd Half Started"),
+    "ET": ("Extra Time", "Extra Time"),
+    "BT": ("Break Time", "Break Time"),
+    "P": ("Penalty", "Penalty In Progress"),
+    "SUSP": ("Suspended", "Match Suspended"),
+    "INT": ("Interrupted", "Match Interrupted"),
+    "FT": ("Full Time", "Match Finished"),
+    "AET": ("Full Time", "Match Finished After Extra Time"),
+    "PEN": ("Full Time", "Match Finished After Penalty"),
+    "PST": ("Postponed", "Match Postponed"),
+    "CANC": ("Cancelled", "Match Cancelled"),
+    "ABD": ("Abandoned", "Match Abandoned"),
+    "AWD": ("Tech Loss", "Technical Loss"),
+    "WO": ("WalkOver", "WalkOver"),
+    "LIVE": ("In Progress", "In Progress"),
+}
+
+def countdown_string(timeIn):
+    duration = timeIn - time.now()
+
+    days = math.floor(duration.hours / 24)
+    hours = math.floor(duration.hours - days * 24)
+    minutes = math.floor(duration.minutes - (days * 24 * 60 + hours * 60))
+    if (days > 0):
+        return str(days) + "d " + str(hours) + "h"
+    elif (hours > 0):
+        return str(hours) + "h " + str(minutes) + "m"
+    else:
+        return str(minutes) + "m"
+
+def get_current_or_next_game_data():
+    cached_data = cache.get("hrbfc-data")
+
+    if cached_data == None:
+        response = http.get(GAME_DATA_PATH).json()
+
+        data = response["message"]
+
+        print("Fetched new data")
+
+        cache.set("hrbfc-data", json.encode(data), ttl_seconds = 55)
+
+        return data
+    else:
+        print("Using cached data")
+        return json.decode(cached_data)
+
+def get_team_names_data():
+    cached_data = cache.get("hrbfc-teams")
+    team_data = {}
+
+    if cached_data == None:
+        response = http.get(TEAM_NAMES_PATH).json()
+
+        print("Fetched team names")
+
+        team_data = response["message"]
+
+        cache.set(
+            "hrbfc-teams",
+            json.encode(team_data),
+            ttl_seconds = 600000,
+        )  # TTL 1 week
+
+    else:
+        print("Using cached team names")
+        team_data = json.decode(cached_data)
+
+    team_data[HRBFC_TEAM_ID] = {
+        "long": "Beavers",
+        "short": "HRB",
+    }
+
+    return team_data
+
+def get_background_image_data():
+    return http.get(BACKGROUND_IMAGE_URL).body()
```

Comment:
Would be good to cache this one as well.

Reply:
And any other HTTP calls without caching I haven't noticed. =)

Reply:
Done